Release v1.10.1

dveeden opened the pull request:
1. I think this should be `v1.10.1`. Or should it be `v1.11.0` ?

2. I used this to create the changelog and then added author links manually

```
$ git log --oneline v1.10.0..HEAD | cut -c9- | sed -E -e 's/^/* /' -e 's%\(#(.*)\)%[#\1](https://github.com/go-mysql-org/go-mysql/pull/\1)%'
* mysql: Fix GetInt() with negative text result [#972](https://github.com/go-mysql-org/go-mysql/pull/972)
* client: Update docs [#970](https://github.com/go-mysql-org/go-mysql/pull/970)
* Cleanup: remove clear_vendor.sh [#971](https://github.com/go-mysql-org/go-mysql/pull/971)
* *: don't use zero value of mysql.Result [#969](https://github.com/go-mysql-org/go-mysql/pull/969)
* mysql: Fix ColumnNumber() panic with nil Resultset [#968](https://github.com/go-mysql-org/go-mysql/pull/968)
* replication: Improve COM_REGISTER_SLAVE [#967](https://github.com/go-mysql-org/go-mysql/pull/967)
* use syscall.Gettimeofday to get current time [#961](https://github.com/go-mysql-org/go-mysql/pull/961)
* Remove duplicate SIGINT in signal handling [#962](https://github.com/go-mysql-org/go-mysql/pull/962)
* feat: Added `time.Time` support for datetime format [#957](https://github.com/go-mysql-org/go-mysql/pull/957)
* mysql field: expose API to modify field vale [#960](https://github.com/go-mysql-org/go-mysql/pull/960)
* replication: Support GTID tag in PreviousGTIDsEvent [#952](https://github.com/go-mysql-org/go-mysql/pull/952)
* zeroalloc: don't use unsafe.Pointer [#959](https://github.com/go-mysql-org/go-mysql/pull/959)
* driver: allow configuration of the driver name [#958](https://github.com/go-mysql-org/go-mysql/pull/958)
* Update README [#950](https://github.com/go-mysql-org/go-mysql/pull/950)
* README: fix example for database/sql [#955](https://github.com/go-mysql-org/go-mysql/pull/955)
* CI: Update Go, MySQL and Ubuntu versions [#954](https://github.com/go-mysql-org/go-mysql/pull/954)
* canal,dump: Improve logging [#953](https://github.com/go-mysql-org/go-mysql/pull/953)
```

3. I assume we create the tag after this is merged? Anything else I need to do?

lance6716 commented:
1. I think the next version can be `v1.11.0`. Currently the patch version is only increased when I did something wrong on `x.y.0` and release a new version without many changes
2. I used `gh` CLI https://github.com/cli/cli to get author names, but I don't remember how to do that. Every time I use chatGPT's help 😂 
3. Yes the only thing to be done is push the git tag after this PR

lance6716 commented on CHANGELOG.md:
```diff
@@ -1,3 +1,22 @@
+## Tag v1.10.1 (2025-01-23)
```
```suggestion
## Tag v1.11.0 (2025-01-23)
```

lance6716 reviewed (commented):
rest lgtm

atercattus commented:
I used a [self-written cli](https://github.com/atercattus/changeloger) to generate the changelog, but it works so-so. I would be glad if we switched or discussed switching to GH releases themselves.

If you need, here's what the utility generates now:
```
### Tag NEW_TAG_HERE (2025-01-23)
* Mysql: Fix GetInt() with negative NEWDECIMAL result. [#972](https://github.com/go-mysql-org/go-mysql/pull/972) ([dveeden](https://github.com/dveeden))
* Client: Update docs. [#970](https://github.com/go-mysql-org/go-mysql/pull/970) ([dveeden](https://github.com/dveeden))
* Cleanup: remove clear_vendor.sh. [#971](https://github.com/go-mysql-org/go-mysql/pull/971) ([dveeden](https://github.com/dveeden))
* *: don't use zero value of mysql.Result. [#969](https://github.com/go-mysql-org/go-mysql/pull/969) ([lance6716](https://github.com/lance6716))
* Mysql: Fix ColumnNumber() panic with nil Resultset. [#968](https://github.com/go-mysql-org/go-mysql/pull/968) ([dveeden](https://github.com/dveeden))
* Replication: Improve COM_REGISTER_SLAVE. [#967](https://github.com/go-mysql-org/go-mysql/pull/967) ([dveeden](https://github.com/dveeden))
* Use syscall.Gettimeofday to get current time. [#961](https://github.com/go-mysql-org/go-mysql/pull/961) ([hjweddie](https://github.com/hjweddie))
* Remove duplicate SIGINT in signal handling. [#962](https://github.com/go-mysql-org/go-mysql/pull/962) ([soya111](https://github.com/soya111))
* Feat: Added `time.Time` support for datetime format. [#957](https://github.com/go-mysql-org/go-mysql/pull/957) ([inoth](https://github.com/inoth))
* Mysql field: expose API to modify field value. [#960](https://github.com/go-mysql-org/go-mysql/pull/960) ([walkline](https://github.com/walkline))
* Replication: Support GTID tag in PreviousGTIDsEvent. [#952](https://github.com/go-mysql-org/go-mysql/pull/952) ([dveeden](https://github.com/dveeden))
* Zeroalloc: don't use unsafe.Pointer. [#959](https://github.com/go-mysql-org/go-mysql/pull/959) ([serprex](https://github.com/serprex))
* Driver: allow configuration of the driver name. [#958](https://github.com/go-mysql-org/go-mysql/pull/958) ([dveeden](https://github.com/dveeden))
* Update README. [#950](https://github.com/go-mysql-org/go-mysql/pull/950) ([dveeden](https://github.com/dveeden))
* README: fix example for database/sql. [#955](https://github.com/go-mysql-org/go-mysql/pull/955) ([dveeden](https://github.com/dveeden))
* CI: Update Go, MySQL and Ubuntu versions. [#954](https://github.com/go-mysql-org/go-mysql/pull/954) ([dveeden](https://github.com/dveeden))
* Canal,dump: Improve logging. [#953](https://github.com/go-mysql-org/go-mysql/pull/953) ([dveeden](https://github.com/dveeden))

```

atercattus reviewed (approved):
Also, we have the last version [here](https://github.com/go-mysql-org/go-mysql/blob/master/README.md?plain=1#L546-L554), but this is only docs.

dveeden commented:
The only issue I see with `gh release create v1.2.3 --generate-notes ...`  etc. is that it doesn't really match the PR workflow of first merging the changelog and then pushing the tag.

dveeden commented:
> Also, we have the last version [here](https://github.com/go-mysql-org/go-mysql/blob/master/README.md?plain=1#L546-L554), but this is only docs.

Thanks. Fixed with a98f973af7b4053f5a37660e3a0950156820acfb

dveeden commented:
> I used a [self-written cli](https://github.com/atercattus/changeloger) to generate the changelog, but it works so-so. I would be glad if we switched or discussed switching to GH releases themselves.
> 
> If you need, here's what the utility generates now:
> 
> ```
> go-mysql-org/go-mysql$ ./changeloger --github-token=ghp_XXX
> ### Tag NEW_TAG_HERE (2025-01-23)
> * Mysql: Fix GetInt() with negative NEWDECIMAL result. [#972](https://github.com/go-mysql-org/go-mysql/pull/972) ([dveeden](https://github.com/dveeden))
> * Client: Update docs. [#970](https://github.com/go-mysql-org/go-mysql/pull/970) ([dveeden](https://github.com/dveeden))
> * Cleanup: remove clear_vendor.sh. [#971](https://github.com/go-mysql-org/go-mysql/pull/971) ([dveeden](https://github.com/dveeden))
> * *: don't use zero value of mysql.Result. [#969](https://github.com/go-mysql-org/go-mysql/pull/969) ([lance6716](https://github.com/lance6716))
> * Mysql: Fix ColumnNumber() panic with nil Resultset. [#968](https://github.com/go-mysql-org/go-mysql/pull/968) ([dveeden](https://github.com/dveeden))
> * Replication: Improve COM_REGISTER_SLAVE. [#967](https://github.com/go-mysql-org/go-mysql/pull/967) ([dveeden](https://github.com/dveeden))
> * Use syscall.Gettimeofday to get current time. [#961](https://github.com/go-mysql-org/go-mysql/pull/961) ([hjweddie](https://github.com/hjweddie))
> * Remove duplicate SIGINT in signal handling. [#962](https://github.com/go-mysql-org/go-mysql/pull/962) ([soya111](https://github.com/soya111))
> * Feat: Added `time.Time` support for datetime format. [#957](https://github.com/go-mysql-org/go-mysql/pull/957) ([inoth](https://github.com/inoth))
> * Mysql field: expose API to modify field value. [#960](https://github.com/go-mysql-org/go-mysql/pull/960) ([walkline](https://github.com/walkline))
> * Replication: Support GTID tag in PreviousGTIDsEvent. [#952](https://github.com/go-mysql-org/go-mysql/pull/952) ([dveeden](https://github.com/dveeden))
> * Zeroalloc: don't use unsafe.Pointer. [#959](https://github.com/go-mysql-org/go-mysql/pull/959) ([serprex](https://github.com/serprex))
> * Driver: allow configuration of the driver name. [#958](https://github.com/go-mysql-org/go-mysql/pull/958) ([dveeden](https://github.com/dveeden))
> * Update README. [#950](https://github.com/go-mysql-org/go-mysql/pull/950) ([dveeden](https://github.com/dveeden))
> * README: fix example for database/sql. [#955](https://github.com/go-mysql-org/go-mysql/pull/955) ([dveeden](https://github.com/dveeden))
> * CI: Update Go, MySQL and Ubuntu versions. [#954](https://github.com/go-mysql-org/go-mysql/pull/954) ([dveeden](https://github.com/dveeden))
> * Canal,dump: Improve logging. [#953](https://github.com/go-mysql-org/go-mysql/pull/953) ([dveeden](https://github.com/dveeden))
> 
> * New release v1.10.0. ...
> * ...
> ```

I've update `CHANGELOG.md` to use that.

The differences with my code are:
```diff
diff --git a/CHANGELOG.md b/CHANGELOG.md
index a32677c..6877f4c 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,21 +1,21 @@
## Tag v1.11.0 (2025-01-23)
* [-mysql:-]{+Mysql:+} Fix GetInt() with negative [-text result-]{+NEWDECIMAL result.+} [#972](https://github.com/go-mysql-org/go-mysql/pull/972) ([dveeden](https://github.com/dveeden))
* [-client:-]{+Client:+} Update [-docs-]{+docs.+} [#970](https://github.com/go-mysql-org/go-mysql/pull/970) ([dveeden](https://github.com/dveeden))
* Cleanup: remove [-clear_vendor.sh-]{+clear_vendor.sh.+} [#971](https://github.com/go-mysql-org/go-mysql/pull/971) ([dveeden](https://github.com/dveeden))
* [-`*`:-]{+*:+} don't use zero value of [-mysql.Result-]{+mysql.Result.+} [#969](https://github.com/go-mysql-org/go-mysql/pull/969) ([lance6716](https://github.com/lance6716))
* [-mysql:-]{+Mysql:+} Fix ColumnNumber() panic with nil [-Resultset-]{+Resultset.+} [#968](https://github.com/go-mysql-org/go-mysql/pull/968) ([dveeden](https://github.com/dveeden))
* [-replication:-]{+Replication:+} Improve [-COM_REGISTER_SLAVE-]{+COM_REGISTER_SLAVE.+} [#967](https://github.com/go-mysql-org/go-mysql/pull/967) ([dveeden](https://github.com/dveeden))
* [-use-]{+Use+} syscall.Gettimeofday to get current [-time-]{+time.+} [#961](https://github.com/go-mysql-org/go-mysql/pull/961) ([hjweddie](https://github.com/hjweddie))
* Remove duplicate SIGINT in signal [-handling-]{+handling.+} [#962](https://github.com/go-mysql-org/go-mysql/pull/962) ([soya111](https://github.com/soya111))
* [-feat:-]{+Feat:+} Added `time.Time` support for datetime [-format-]{+format.+} [#957](https://github.com/go-mysql-org/go-mysql/pull/957) ([inoth](https://github.com/inoth))
* [-mysql-]{+Mysql+} field: expose API to modify field [-vale-]{+value.+} [#960](https://github.com/go-mysql-org/go-mysql/pull/960) ([walkline](https://github.com/walkline))
* [-replication:-]{+Replication:+} Support GTID tag in [-PreviousGTIDsEvent-]{+PreviousGTIDsEvent.+} [#952](https://github.com/go-mysql-org/go-mysql/pull/952) ([dveeden](https://github.com/dveeden))
* [-zeroalloc:-]{+Zeroalloc:+} don't use [-unsafe.Pointer-]{+unsafe.Pointer.+} [#959](https://github.com/go-mysql-org/go-mysql/pull/959) ([serprex](https://github.com/serprex))
* [-driver:-]{+Driver:+} allow configuration of the driver [-name-]{+name.+} [#958](https://github.com/go-mysql-org/go-mysql/pull/958) ([dveeden](https://github.com/dveeden))
* Update [-README-]{+README.+} [#950](https://github.com/go-mysql-org/go-mysql/pull/950) ([dveeden](https://github.com/dveeden))
* README: fix example for [-database/sql-]{+database/sql.+} [#955](https://github.com/go-mysql-org/go-mysql/pull/955) ([dveeden](https://github.com/dveeden))
* CI: Update Go, MySQL and Ubuntu [-versions-]{+versions.+} [#954](https://github.com/go-mysql-org/go-mysql/pull/954) ([dveeden](https://github.com/dveeden))
* [-canal,dump:-]{+Canal,dump:+} Improve [-logging-]{+logging.+} [#953](https://github.com/go-mysql-org/go-mysql/pull/953) ([dveeden](https://github.com/dveeden))

## Tag v1.10.0 (2024-11-20)
* replication,cmd: improve flavor handling [#946](https://github.com/go-mysql-org/go-mysql/pull/946) ([dveeden](https://github.com/dveeden))
```

Which is because:
- Uppercase the first letter (I don't think we should as I often use the `<package,...>: <description>` format)
- Add `.` at the end of the title
- This still need manual changes for `*: ...` as title to not break formatting.

dveeden commented:
And as GitHub doesn't format/colors word diffs in a useful way:
![image](https://github.com/user-attachments/assets/96560b9a-b670-44a6-82f6-aec92e5f5434)

lance6716 reviewed (approved):
lgtm. I just loop each PR number and use `gh` to query PR metadata to get the author.
